starboard/tools/testing/test_runner.py

```diff
@@ -163,12 +163,24 @@ class TestLauncher(object):
 
   def __init__(self, launcher):
     self.launcher = launcher
-    self.runner_thread = threading.Thread(target=self._Run)
 
     self.return_code_lock = threading.Lock()
     self.return_code = 1
 
-  def Start(self):
+  def Start(self, extended, deploy):
+    if extended:
```

Comment:
Do we actually need `extended` as a parameter, or can we just rely on `hasattr(self.launcher, Run2)`  ? 

I'f maybe move this into the launcher code so that you could cal `self.launcher.hasExtendedInterface()` which checks if the launcher does have all required args. Maybe even `self.launcher.hasInterfaceVersion(2)` which checks for presence of `Run2` and `InitDevice` and `Deploy` ( and perhaps `Kill`, see below)

Reply:
I removed `extended` from parameters. Now `test_runner` always uses `Run2`, if implemented. Yet, I see no need in `hasExtendedInterface` method.